beacon-chain/rpc/prysm/v1alpha1/validator/sync_committee.go

```diff
@@ -0,0 +1,186 @@
+package validator
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// GetSyncMessageBlockRoot retrieves the sync committee block root of the beacon chain.
+func (vs *Server) GetSyncMessageBlockRoot(
+	ctx context.Context, _ *emptypb.Empty,
+) (*ethpb.SyncMessageBlockRootResponse, error) {
+	r, err := vs.HeadFetcher.HeadRoot(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not retrieve head root: %v", err)
+	}
+
+	return &ethpb.SyncMessageBlockRootResponse{
+		Root: r,
+	}, nil
+}
+
+// SubmitSyncMessage submits the sync committee message to the network.
+// It also saves the sync committee message into the pending pool for block inclusion.
+func (vs *Server) SubmitSyncMessage(ctx context.Context, msg *ethpb.SyncCommitteeMessage) (*emptypb.Empty, error) {
+	errs, ctx := errgroup.WithContext(ctx)
+
+	idxResp, err := vs.syncSubcommitteeIndex(ctx, msg.ValidatorIndex, msg.Slot)
```

Comment:
Would be good to look at how many calls this triggers and perhaps could use a cache here

Reply:
This is pretty well cached underneath. In the current spec there's only 512 sync committee participants active at a given time. Chances are low to pursue further overish optimization